service/src/main/java/com/myhome/configuration/EmailTemplateConfig.java

```diff
@@ -0,0 +1,57 @@
+package com.myhome.configuration;
+
+import com.myhome.configuration.properties.mail.MailProperties;
+import lombok.RequiredArgsConstructor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.support.ResourceBundleMessageSource;
+import org.thymeleaf.spring5.SpringTemplateEngine;
+import org.thymeleaf.templateresolver.ClassLoaderTemplateResolver;
+import org.thymeleaf.templateresolver.ITemplateResolver;
+
+import java.util.Locale;
+
+@Configuration
+@RequiredArgsConstructor
+public class EmailTemplateConfig {
+
+  private final MailProperties mailProperties;
+
+  @Bean
+  public ITemplateResolver thymeleafTemplateResolver() {
+    MailProperties.MailTemplate templatesProperties = mailProperties.getTemplate();
+
+    ClassLoaderTemplateResolver templateResolver = new ClassLoaderTemplateResolver();
+
+    String templatePath = templatesProperties.getPath();
+    String fileSeparator = System.getProperty("file.separator");
+    templateResolver.setPrefix(templatePath.endsWith(fileSeparator) ? templatePath : templatePath + fileSeparator);
+
+    templateResolver.setSuffix(templatesProperties.getFormat());
+    templateResolver.setTemplateMode(templatesProperties.getMode());
+    templateResolver.setCharacterEncoding(templatesProperties.getEncoding());
+    templateResolver.setCacheable(templatesProperties.isCache());
+    return templateResolver;
+  }
+
+  @Bean
+  public ResourceBundleMessageSource emailMessageSource() {
+    MailProperties.MailTemplatesLocalization mailLocalization = mailProperties.getLocalization();
+
+    ResourceBundleMessageSource messageSource = new ResourceBundleMessageSource();
+    messageSource.setBasename(mailLocalization.getPath());
+    messageSource.setDefaultLocale(Locale.ENGLISH);
+    messageSource.setDefaultEncoding(mailLocalization.getEncoding());
+    messageSource.setCacheSeconds(mailLocalization.getCacheSeconds());
+    return messageSource;
+  }
+
+  @Bean
+  public SpringTemplateEngine thymeleafTemplateEngine(ResourceBundleMessageSource emailMessageSource, ITemplateResolver thymeleafTemplateResolver) {
```

Comment:
Does `ResourceBundleMessageSource Resolver` and `ITemplateResolver ` needs to be a Bean?
Isn't enough to have it as follows:

```
  @Bean
  public SpringTemplateEngine thymeleafTemplateEngine() {
    SpringTemplateEngine templateEngine = new SpringTemplateEngine();
    templateEngine.setTemplateResolver(thymeleafTemplateResolver());
    templateEngine.setTemplateEngineMessageSource(emailMessageSource());
    return templateEngine;
  }

  private ITemplateResolver thymeleafTemplateResolver() {
    ClassLoaderTemplateResolver templateResolver = new ClassLoaderTemplateResolver();

    String templatePath = emailTemplateProperties.getPath();
    String fileSeparator = System.getProperty("file.separator");
    templateResolver.setPrefix(templatePath.endsWith(fileSeparator) ? templatePath : templatePath + fileSeparator);

    templateResolver.setSuffix(emailTemplateProperties.getFormat());
    templateResolver.setTemplateMode(emailTemplateProperties.getMode());
    templateResolver.setCharacterEncoding(emailTemplateProperties.getEncoding());
    templateResolver.setCacheable(emailTemplateProperties.isCache());
    return templateResolver;
  }

  private ResourceBundleMessageSource emailMessageSource() {
    ResourceBundleMessageSource messageSource = new ResourceBundleMessageSource();
    messageSource.setBasename(emailTemplateLocalization.getPath());
    messageSource.setDefaultLocale(Locale.ENGLISH);
    messageSource.setDefaultEncoding(emailTemplateLocalization.getEncoding());
    messageSource.setCacheSeconds(emailTemplateLocalization.getCacheSeconds());
    return messageSource;
  }
  
  ```

Reply:
`ResourceBundleMessageSource` using to localize mail subjects in `MailSDJpaService`.
`ITemplateResolver` need to get `htmlBody` of mail
`messageSource` here:
```
private String getLocalizedMessage(String prop) {
    String message = "";
    try {
      message = messageSource.getMessage(prop, null, LocaleContextHolder.getLocale());
    } catch (Exception e) {
      message = prop + ": localization error";
    }
    return message;
  }
```
and `emailTemplateEngine` here:
```
private boolean send(String emailTo, String subject, String templateName, Map<String, Object> templateModel) {
    try {
      Context thymeleafContext = new Context(LocaleContextHolder.getLocale());
      thymeleafContext.setVariables(templateModel);
      String htmlBody = emailTemplateEngine.process(templateName, thymeleafContext);
      sendHtmlMessage(emailTo, subject, htmlBody);
    } catch (MailException | MessagingException mailException) {
      log.error("Mail send error!", mailException);
      return false;
    }
    return true;
  }
```